api/nodes/views.py

```diff
@@ -467,9 +700,108 @@ def get_default_queryset(self):
             contributors.append(contributor)
         return contributors
 
-    # overrides ListAPIView
+    # overrides ListBulkCreateUpdateDestroyAPIView
     def get_queryset(self):
-        return self.get_queryset_from_request()
+        queryset = self.get_queryset_from_request()
+
+        # If bulk request, queryset only contains contributors in request
+        if isinstance(self.request.data, list):
+            contrib_ids = [item['id'] for item in self.request.data]
+            queryset[:] = [contrib for contrib in queryset if contrib._id in contrib_ids]
+
+        return queryset
+
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def get_serializer(self, *args, **kwargs):
+        """
+         Adds many=True to serializer if bulk operation.
+        """
+
+        if "data" in kwargs:
+            data = kwargs["data"]
+
+            if isinstance(data, list):
+                kwargs['many'] = True
+
+        return super(NodeContributorsList, self).get_serializer(*args, **kwargs)
+
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def get_serializer_class(self):
+        """
+        Use NodeContributorDetailSerializer which requires 'id'
+        """
+        if self.request.method == 'PUT' or self.request.method == 'PATCH' or self.request.method == 'DELETE':
+            return NodeContributorDetailSerializer
+        else:
+            return NodeContributorsSerializer
+
+    # overrides ListBulkCreateUpdateDestroyView
+    def create(self, request, *args, **kwargs):
+        """
+        Correctly formats both bulk and single POST response
+        """
+        response = super(NodeContributorsList, self).create(request, *args, **kwargs)
+        if 'data' in response.data:
+            return response
+        return Response({'data': response.data}, status=status.HTTP_201_CREATED)
+
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def bulk_update(self, request, *args, **kwargs):
+        """
+        Correctly formats bulk PUT/PATCH response
+        """
+        response = super(NodeContributorsList, self).bulk_update(request, *args, **kwargs)
+        return Response({'data': response.data}, status=status.HTTP_200_OK)
+
+    # Overrides ListBulkCreateUpdateDestroyAPIView
+    def bulk_destroy(self, request, *args, **kwargs):
+        """
+        Handles bulk destroy of contributors. Handles validation and enforces bulk limit.
+        """
+        contrib_list = []
+        object_type = 'contributors'
```

Comment:
Rather than hardcoding these, we should be able to get the type from the view's Serializer. This could happen in a custom base class, as suggested in a previous comment.